scrapy/core/spidermw.py

```diff
@@ -60,8 +60,18 @@ def process_spider_exception(_failure):
             return _failure
 
         def process_spider_output(result):
+            def wrapper(result_iterable):
+                try:
+                    for r in result_iterable:
+                        yield r
+                except Exception as ex:
+                    exception_result = process_spider_exception(Failure(ex))
```

Comment:
Exceptions from `process_spider_output` are handled only when the method returns a generator. There are two cases:
- `process_spider_output` is _not_ a generator. In that case, if an exception is raised the function's return values does not pass along, and since there is no `result_iterable` the exception handler does not get called. This is exactly the same as the current behaviour and documentation.
- `process_spider_output` is a generator. In that case, the exception is not raised right away, instead it actually raises when iterating over `result_iterable`, and the exception handler is called. This is what could backwards compability, but since it's the very thing that's currently broken, I don't think anyone is relying on this functionality.

@kmike Please check the above to see if it makes any sense :-P